File: ide/src/main/java/org/dau/ide/schema/SchemaDemoBlock.java
package org.dau.ide.schema;

import org.dau.runtime.Block;

@Block("Demo block")
public class SchemaDemoBlock {

  public SchemaDemoBlock(String sync, String name, int code, double len, double angle, double speed) {
  }

  public double distance() {
    return 0d;
  }

  public double out() {
    return 0d;
  }

  public double delta() {
    return 0;
  }
}

File: ide/src/main/kotlin/org/dau/ide/schema/SchemaDemoBlock.kt
package org.dau.ide.schema

import org.dau.runtime.Block

@Block("Demo block")
class SchemaDemoBlock(sync: String, name: String, code: Int, len: Double, angle: Double, speed: Double) {

  fun distance(): Double {
    return 0.0
  }

  fun out(): Double {
    return 0.0
  }

  fun delta(): Double {
    return 0.0
  }
}
